src/consensus.rs

```diff
@@ -93,104 +96,102 @@ impl<T: Proposition> Consensus<T> {
     // membership: gen = pending_gen
     /// Handles a signed vote
     /// Returns the vote we cast and the reached consensus vote in case consensus was reached
-    pub fn handle_signed_vote(
-        &mut self,
-        signed_vote: SignedVote<T>,
-        gen: Generation,
-    ) -> Result<VoteResponse<T>> {
+    pub fn handle_signed_vote(&mut self, signed_vote: SignedVote<T>) -> Result<VoteResponse<T>> {
+        info!("[MBR-{}] handling vote {:?}", self.id(), signed_vote);
         if let Err(faults) = self.detect_byzantine_voters(&signed_vote) {
-            println!("[MBR-{}] Found faults {:?}", self.id(), faults);
+            info!("[MBR-{}] Found faults {:?}", self.id(), faults);
             self.faults.extend(faults);
         }
+        if self.faults.contains_key(&signed_vote.voter) {
+            info!("[MBR-{}] dropping vote from faulty voter", self.id());
+            return Ok(VoteResponse::WaitingForMoreVotes);
+        }
```

Comment:
So my understanding is that: 
- we ignore new votes from faulty voters
- we update our blacklist with the sender's every time we receive a new vote
- we send out our blacklist along with our votes

But it seems that older votes from a faulty node (those received and logged before they where added to our blacklist) are kept. And we even count them in when calculating SM or split situations. Did I understand it correctly? If so is that a feature?